compartmentalize-grains-via-pop.md

```diff
@@ -0,0 +1,71 @@
+- Feature Name: compartmentalize-grains-via pop
+- Start Date: 2020-06-22
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Grains is currently a static component of salt.  We want it to be developed in a more compartmentalized fashion.
+The engineers at saltstack have created kernel-specific projects that functionally replace the grains
+provided by salt with POP.  We propose that this project, codenamed `corn` replaces grains in salt while
+maintaining backwards compatibility.
+
+# Motivation
+[motivation]: #motivation
+
+"The original implementation of grains was not designed to scale to the level that salt requires" - thatch45
+Tom wrote POP to solve this problem and we're going to use it. 
+
+By using Plugin Oriented Programming, grains can be separated into individual projects that can be maintained
+by the community safely.
+
+This also solves the issue of `core.py` being an unmaintainable monolith.  It allows platform specific grains
+to be independently maintained.
```

Comment:
So could simply not having them all in a single file.

Reply:
:point_up: 

I would be curious to see a comparison between these approaches.

Reply:
It's also about having grainsv2 be tested thoroughly on each platform, having a much faster cadence/release cycle for grains, having working group/community maintained projects that implement platform-specific grains/states/exec modules, speeding up the test suite, speeding up grains.... Breaking up core.py is just one piece of the puzzle